### PR TITLE
Avoid file locks in `LauncherTest.remotable`

### PR DESCRIPTION
**Pull request by jglick**
Amending test introduced in #3563 since as noted in https://github.com/jenkinsci/jenkins/pull/8006#issuecomment-1832699830 it can have file locking issues on Windows. Simplified as per https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/21/commits/4cf6db4e8ab2e3c22bda7a66c958afaec465768c.

### Testing done

Passes locally on both Linux and Windows. (Could not reproduce failure on Windows but did not try hard.)

### Proposed changelog entries

- N/A

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]
Nice!

**Review by NotMyFault — APPROVED** [review]

**Comment by NotMyFault** [comment]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
